### PR TITLE
optimizations

### PR DESCRIPTION
**Pull request by gdikarev**

### CODE REVIEW

File: src/components/Registration/UserData.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+function getRadioBoxStyle(isSelected: boolean) {
+    return {
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        fontWeight: isSelected ? 'bold' : 'normal',
+        color: isSelected ? '#000' : '#666',
+    };
+}
```
Comment (vlad-kk) [inline src/components/Registration/UserData.tsx:140]: move to separate file. check how implemet behavior using emotion/css

File: src/pages/Registration/Registration.tsx [hunk 2]
```diff
@@ -88,35 +94,27 @@ const RegistrationPage: FC = () => {
         return () => {
             mainButton.off('click', handleClick);
         };
-    }, [mainButton]);
+    }, [currentStepIndex]);
 
     useEffect(() => {
-        if (!isFirstStep) backButton.show()
+        if (!isFirstStep) backButton.show();
```
Comment (vlad-kk) [inline src/pages/Registration/Registration.tsx:100]: i prefer code like` !isFirstStep && backButton.show();`
but it's ok too. if you decide to change, do it in all places

File: src/shared/type/userRegistrationData.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ export type UserRegistrationData = {
     gender: string,
     search_gender: string,
     birthday: string,
-    locale: string,
+    locale: string | undefined,
```
Comment (vlad-kk) [inline src/shared/type/userRegistrationData.ts:8]: u can write like` locale?: string;`

it's the same `locale: string | undefined ===  locale?: string;`

**Review by vlad-kk — APPROVED** [review]

File: src/components/Registration/UserData.tsx (hunk 1 above)
Comment (gdikarev) [reply src/components/Registration/UserData.tsx:140]: will move on later stage. I will split this component more

**Pull request merged**